solutions/standard/variables.tf

```diff
@@ -0,0 +1,310 @@
+      name      = string
+      direction = optional(string, "inbound")
+      remote    = string
+      tcp = optional(
+        object({
+          port_max = optional(number)
+          port_min = optional(number)
+        })
+      )
+      udp = optional(
+        object({
+          port_max = optional(number)
+          port_min = optional(number)
+        })
+      )
+      icmp = optional(
+        object({
+          type = optional(number)
+          code = optional(number)
+        })
+      )
+    })
+  )
+
+  validation {
+    error_message = "Security group rule direction can only be `inbound` or `outbound`."
+    condition = (var.security_group_rules == null || length(var.security_group_rules) == 0) ? true : length(distinct(
+      flatten([
+        # Check through rules
+        for rule in var.security_group_rules :
+        # Return false if direction is not valid
+        false if !contains(["inbound", "outbound"], rule.direction)
+      ])
+    )) == 0
+  }
+
+  validation {
+    error_message = "Security group rule names must match the regex pattern ^([a-z]|[a-z][-a-z0-9_]*[a-z0-9])$."
+    condition = (var.security_group_rules == null || length(var.security_group_rules) == 0) ? true : length(distinct(
+      flatten([
+        # Check through rules
+        for rule in var.security_group_rules :
+        # Return false if direction is not valid
+        false if !can(regex("^([a-z]|[a-z][-a-z0-9_]*[a-z0-9])$", rule.name))
+      ])
+    )) == 0
+  }
+
+  validation {
+    error_message = "Security group rules can only have one of `icmp`, `udp`, or `tcp`."
+    condition = (var.security_group_rules == null || length(var.security_group_rules) == 0) ? true : length(distinct(
+      # Get flat list of results
+      flatten([
+        # Check through rules
+        for rule in var.security_group_rules :
+        # Return true if there is more than one of `icmp`, `udp`, or `tcp`
+        true if length(
+          [
+            for type in ["tcp", "udp", "icmp"] :
+            true if rule[type] != null
+          ]
+        ) > 1
+      ])
+    )) == 0 # Checks for length. If all fields all correct, array will be empty
+  }
+}
+
+variable "vpn_client_access_group_users" {
+  description = "The list of users in the Client to Site VPN Access Group"
+  type        = list(string)
+  default     = []
+}
+
+variable "access_group_name" {
+  type        = string
+  description = "The name of the IAM Access Group to create if the 'create_policy' input variable is `true`."
+  default     = "client-to-site-vpn-access-group"
+}
+
+variable "create_policy" {
+  description = "Whether to create a new access group (using the value of the 'access_group_name' input variable) with a VPN Client role."
+  type        = bool
+  default     = true
+}
+
+variable "vpn_server_routes" {
+  type = map(object({
+    destination = string
+    action      = string
+  }))
+  description = "A map of server routes to be added to created VPN server."
```

Comment:
similar to above comment - should we really be defaulting these? We should discuss with the team